test/Collector.test.ts

```diff
@@ -0,0 +1,185 @@
+import { expect, use } from 'chai';
+import Collector from '../build/contracts/Collector.json';
+import TestToken from '../build/contracts/TestToken.json';
+import {
+    deployContract,
+    loadFixture,
+    MockProvider,
+    solidity
+} from 'ethereum-waffle';
+import { BaseContract } from '@ethersproject/contracts';
+import { Wallet } from '@ethersproject/wallet';
+
+use(solidity);
+
+const buildPartners = (
+    wallets: Wallet[],
+    share1?: number,
+    share2?: number,
+    share3?: number,
+    share4?: number
+) => {
+    const [owner, remainder, partner1, partner2, partner3, partner4] = wallets;
+
+    return [
+        {
+            beneficiary: partner1.getAddress(),
+            share: share1 ?? 20
+        },
+        {
+            beneficiary: partner2.getAddress(),
+            share: share2 ?? 35
+        },
+        {
+            beneficiary: partner3.getAddress(),
+            share: share3 ?? 13
+        },
+        {
+            beneficiary: partner4.getAddress(),
+            share: share4 ?? 32
+        }
+    ];
+};
+
+async function deployCollector(
+    wallets: Wallet[],
+    testToken: BaseContract,
+    partners: any
+) {
+    const [ownerWallet, remainderWallet] = wallets;
+
+    const collector = await deployContract(ownerWallet, Collector, [
+        ownerWallet.address,
+        testToken.address,
+        partners,
+        remainderWallet.address
+    ]);
+
+    return { collector };
+}
+describe('Collector', function () {
+    async function deployTokenFixture(wallets) {
+        const testToken = await deployContract(wallets[0], TestToken);
```

Comment:
It's not clear what's in the 0 position of this array. Perhaps using a constant could be useful. Something like wallets[TOKEN_POSITION] or wallets[OWNER_POSITION] and at top of the file or in a separate one: OWNER_POSITION=0;
Otherwise the positions of the array can be documented on a comment. 
See [https://refactoring.guru/replace-magic-number-with-symbolic-constant](url) and [https://stackoverflow.com/questions/2089429/magic-numbers-in-arrays-c](url)

Reply:
Agree on being more explicit. Will use constants instead.